src/OrbitGl/TrackContainer.cpp

```diff
@@ -370,11 +369,66 @@ void TrackContainer::SetIteratorOverlayData(
   RequestUpdate(RequestUpdateScope::kDraw);
 }
 
+void TrackContainer::DrawThreadDependency(PrimitiveAssembler& primitive_assembler,
+                                          PickingMode picking_mode) {
+  std::optional<orbit_client_data::ThreadStateSliceInfo> thread_state_slice =
+      app_->selected_thread_state_slice();
+  if (!thread_state_slice.has_value() || picking_mode != PickingMode::kNone) {
+    return;
+  }
+
+  orbit_client_data::ThreadID start_arrow_thread = thread_state_slice->wakeup_tid();
+  orbit_client_data::ThreadID end_arrow_thread = thread_state_slice->tid();
+
+  std::optional<ThreadTrack*> temp_start_arrow_track =
+      track_manager_->GetThreadTrack(start_arrow_thread);
+  std::optional<ThreadTrack*> temp_end_arrow_track =
+      track_manager_->GetThreadTrack(end_arrow_thread);
+
+  if (!temp_end_arrow_track.has_value() || !temp_start_arrow_track.has_value()) {
+    return;
+  }
+
+  ThreadTrack* start_arrow_track = temp_start_arrow_track.value();
+  ThreadTrack* end_arrow_track = temp_end_arrow_track.value();
+
+  const std::vector<Track*>& threads = track_manager_->GetVisibleTracks();
+
+  if (std::find(threads.begin(), threads.end(), start_arrow_track) == threads.end() ||
+      std::find(threads.begin(), threads.end(), end_arrow_track) == threads.end()) {
+    return;
+  }
+
+  const Color kArrowColor(255, 255, 255, 255);
+  const float kOffsetToGetCenterOfEndingThreadStateBar =
+      end_arrow_track->GetThreadStateBarHeight() / 2;
+  const float kOffsetToGetCenterOfStartingThreadStateBar =
+      start_arrow_track->GetThreadStateBarHeight() / 2;
+  float x = timeline_info_->GetWorldFromTick(thread_state_slice->begin_timestamp_ns());
```

Comment:
`end_arrow_x`

Reply:
No need for this since the arrow has only one x. Both the start and end have the same x so it doesn't make sense to call it end_arrow_x since it's also start_arrow_x.